File: DataBindingObservableJava/app/src/main/java/com/websarva/wings/android/databindingobservablejava/data/model/Rectangle.java
package com.websarva.wings.android.databindingobservablejava.data.model;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

import com.websarva.wings.android.databindingobservablejava.BR;

public class Rectangle extends BaseObservable {
	private int _height = 0;
	private int _width = 0;

	public int getArea() {
		return _height * _width;
	}

	@Bindable
	public String getAreaStr() {
		return String.valueOf(getArea());
	}

	public void setHeight(int height) {
		_height = height;
		notifyPropertyChanged(BR.heightStr);
		notifyPropertyChanged(BR.areaStr);
	}

	public void setWidth(int width) {
		_width = width;
		notifyPropertyChanged(BR.widthStr);
		notifyPropertyChanged(BR.areaStr);
	}

	@Bindable
	public String getHeightStr() {
		return String.valueOf(_height);
	}

	@Bindable
	public String getWidthStr() {
		return String.valueOf(_width);
	}
}

File: DataBindingLiveDataKotlin/app/src/main/java/com/websarva/wings/android/databindinglivedatakotlin/data/model/Rectangle.kt
package com.websarva.wings.android.databindinglivedatakotlin.data.model

data class Rectangle(val height: Int, val width: Int) {
	fun getArea(): Int {
		return height * width
	}
	fun getAreaStr(): String {
		return getArea().toString()
	}

	fun getHeightStr(): String {
		return height.toString()
	}

	fun getWidthStr(): String {
		return  width.toString()
	}
}
